cypress-io/cypress
# fix: only show active router link styles on parent page

- Closes #21873

### User facing changelog

Minor UI tweak to better indicate how to return to the Specs list when the test runner is active

### Additional details

During the bug hunt, several Cypress employees on a video call could not see how to get back to the specs list from the runner, which is fair since the link that navigates there is active, so it is not styled to be clickable (other links have cursor: pointer and hover states, this does not, since we are already inside that part of the app). 

This change makes it so the navigation links are only styled as active when the URL is an exact match. Now, if the user goes to a child page (i.e. test runner is a child page of the Specs page) the link to the parent page will appear clickable as it no longer has active styles.

### Steps to test

1. `yarn dev`
2. Select "E2E testing"
3. Confirm that the "Specs" nav link has active styles (purple color, default cursor on hover)
4. Click on an E2E spec, i.e. `logger.cy.ts`
5. Confirm that the "Specs" nav link no longer has active styles (gray color, pointer cursor on hover)
6. Confirm that clicking the "Specs" nav link returns the user to the "Specs" parent page

### How has the user experience changed?

**Before:**
<img width="515" alt="Screen Shot 2022-06-02 at 4 45 10 PM" src="https://user-images.githubusercontent.com/26726429/171758219-90a7bb16-fd5a-4de9-8749-0a8f95a53dfe.png">

**After:**
<img width="515" alt="Screen Shot 2022-06-02 at 4 45 28 PM" src="https://user-images.githubusercontent.com/26726429/171758150-ccfc8069-2115-4299-ac8b-f8edbadc4fe4.png">


### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
